### ISSUE TITLE
Connect invitation fails

### ISSUE THREAD

**Issue by rowanshedden** [openwallet-foundation/bifold-wallet#367]
When I scan a QR Code set up with a connect invitation the connection conversation just dies after a trust ping is sent. An earlier version of the aries-mobile-agent-react-native project did not have this issue. It has only started happening with this latest release.

This is the ACA-py Agent log snippet:
```sh
2022-07-07 06:29:42,988 aries_cloudagent.connections.base_manager WARNING No corresponding DID found for sender verkey: 5FesrZZc4XFGijm9xFRSo7B5iBUajigDGW5zPcWjFpfT
2022-07-07 06:29:43,003 aries_cloudagent.connections.base_manager WARNING No corresponding DID found for recipient verkey: GMFKiwWqCoM4q9q4BRouu5jRBGiVLd42G6dQ6Pxb46L2
2022-07-07 06:29:43,112 aiohttp.access INFO 172.31.70.15 [07/Jul/2022:06:29:42 +0000] “POST / HTTP/1.1” 200 2617 “-” “AriesBifold/1 CFNetwork/1333.0.4 Darwin/21.5.0”
2022-07-07 06:29:43,775 aries_cloudagent.messaging.base_handler INFO Received trust ping from: 8oYgLi4QGwRi4eA25J1CKM
```
 
As you can see the last thing received is that trust ping….
 
 
This is the Mediator log snippet:
```sh
 DEBUG  DEBUG: Pack outbound message https://didcomm.org/trust_ping/1.0/ping
 DEBUG  DEBUG: Sending outbound message to endpoint ‘https://aca-staging.sitalab.io/’ {
  “payload”: {
    “protected”: “eyJlbmMiOiJ4Y2hhY2hhMjBwb2x5MTMwNV9pZXRmIiwidHlwIjoiSldNLzEuMCIsImFsZyI6IkF1dGhjcnlwdCIsInJlY2lwaWVudHMiOlt7ImVuY3J5cHRlZF9rZXkiOiJlSTUyZjJZTThya3dSRXdaQm1UUmNicDU1Zy1fSzVWaTYzZnVGbUh1WEYyM3dqY2tOWnlRRWdkZVNHekVXWHlrIiwiaGVhZGVyIjp7ImtpZCI6IjhoNm9BYllTTllRWkw3WExLdFBZaW1UNDFDUEpBTHk3a1RmZUNGbVJnNmJQIiwiaXYiOiJxSG1OX01fMUFwdExhVVJDODBpcUpKSjh4Wl9aYW44QSIsInNlbmRlciI6IjR0TUZMa3pZTmk0RDgwTGVqYXhyMGRlbTZXUXRsS1JJdWcwR2JESkhOUTVKanF5S2hoTDdhcV96S1ozOFVNNmRiNUxtQy1iUjgtTzhRREJicVJhXzhnZ1Zaa2VaWThucVd5YnpIZmdkTENPRWYyMHdKdElkNlBtV1hpaz0ifX1dfQ==“,
    “iv”: “_b23lHT8XuQZ8OON”,
    “ciphertext”: “sP2WM4lOoaep2WdTAeBA6JLouNWAL8P-BKG_px7Dm5Uwv3Gzo3Z1TQYbWl5pnuoopAIKQxm-qwVXKTk0H47G_CMGB0Fs7OuxGGb6V_MquogJHSdOWc2Hi3v6xm03RjA-fXYRtTSWvWYEMT0JyUaQr5W4isujGUfkqcdE8K1OEQY6orfMt0aCzt2ffND4cSidudh32iMWuVuO0wKBhqOd”,
    “tag”: “rTJOME5WWjkQZtxKIFLEfQ==”
  }
}
 DEBUG  DEBUG: Request was aborted due to timeout. Not throwing error due to return routing on sent message
 DEBUG  DEBUG: No response received.
```

After that the app just spins, waiting for connect completion.

Any idea what has been changed in the processing so that it no longer works?

**Comment by amanji** [comment]
To confirm, this is using the latest build on the `main` branch correct?

**Comment by rowanshedden** [comment]
Correct, this is for the latest build on the main branch.

After some further investigation it appears those log snippets are misleading and actually have no relevance to the issue. I have an earlier build of BiFold, where the connection, after the QR scan, is complete and the app returns to the Home page as expected. However, in this latest version the app just keeps spinning, after the QR scan, with a message that the response is taking too long. Yet when I go to the Contacts page I can see a successful connection with status "complete".

This suggests to me that the app is not detecting the connection "complete" status and therefore is not returning to the Home page as it should. I am trying to find where that happens in the code, but as you are more familiar with it than me and can suggest where to look?

**Comment by itsdheerajbudhiraja** [comment]
I am facing same issue in react-native environment (Not reproducible in nodeJS env).

No logs of Message received / processed from mediator. Socket onmessage itself is not triggering itself. While mediator (ACA-py) logs says Message written to socket. 

```
 DEBUG  DEBUG: Pack outbound message https://didcomm.org/trust_ping/1.0/ping
 DEBUG  DEBUG: Sending outbound message to endpoint 'https://ariesagent.com:10000' {
  "payload": {
    "protected": "eyJlbmMiOiJ4Y2hhY2hhMjBwb2x5MTMwNV9pZXRmIiwidHlwIjoiSldNLzEuMCIsImFsZyI6IkF1dGhjcnlwdCIsInJlY2lwaWVudHMiOlt7ImVuY3J5cHRlZF9rZXkiOiJnaEQwWnZ6eU1uWVVOM1Z1MWYwTHdLSFU5blprYW54MzhxYU9CLWFxbGdUSlJ2MjhkT1JPQ2lWN1VxUGlfc0lZIiwiaGVhZGVyIjp7ImtpZCI6IkJjeXREczV2V3lKQ0hyM0ZoUXNQa3RiSEdUczlrVXZmWmtEcXFrRzVhVFVhIiwiaXYiOiJsSGViODZpcVZudzdnSEJaczNTeWZtMURrOHMzSzNYdCIsInNlbmRlciI6IkpBVVNyRi1JakJLdWtmejV4UFdHbmxDVF9fcDNnQ1l5Q2UxZnFPR0g3RGEySFZUYmZidEVqRWIzckNXZXVIRUxTMnBpT3JocHNSOUU0UmhoSGVqZk42NDRBTGtReHhET2NBOXZ2QjJfX2R0dktFenBiR3drZ1F5QUpnPT0ifX1dfQ==",
    "iv": "ZD6SJUQr9OTy-2q7",
    "ciphertext": "CulhCK-3JL69WcHT0fOaiNq3OePwVZoyYNMy1HpvqFNiDBDumIZIBeE7Jax8YiPIVmJ8z3y5oWRAnUrdaX7OfL1t11U1ZuXA7KbjOaHG1f3POJKvjzqaA6Nzp1sUCuj342kIC79JZyjdBEPoBWszVDWxu03a0JAf3WLP",
    "tag": "v6_mMN0oJ-WmybqKT4HOyQ=="
  }
}
 DEBUG  DEBUG: No response received.
```

**Comment by amanji** [comment]
May I suggest updating to the latest version of the wallet that uses AFJ 0.2.0, unless you have already done so? There could be number of causes here but it looks very similar to one we previously observed when using a mediator via websockets. If not we may need to gather some more information to try and replicate or pinpoint the issue.

**Comment by itsdheerajbudhiraja** [comment]
@amanji  I am facing issues with AFJ 0.2.0 only after upgrading my app. I also tries Aries Bifold latest from main branch (yesterday), faced same issue. Mediator messages are highly inconsistent. (Sometime message comes but not processed by Agent.event.observable, while most of times message itself is not read by websocket) - While mediator logs are normal and says - Message written to socket (Same logs in working app of AFJ1). Same mediator we are using and same logs - So mediator seems to be fine. 

Issue lies somewhere in React native websocket class Inconsistency as In nodeJS env, AFJ2.0, I was not able to reproduce till now. (I am testing on Android 12, Versions in my app are same as of Aries Bifold). 

More details and all logs I had shared on Discord earlier: 
https://discord.com/channels/905194001349627914/990995367149588511

**Comment by JamesKEbert** [comment]
I worked some with Rowan on his issue (unsure if it's related to @itsdheerajbudhiraja's problem though). It is related to how not all use cases immediately follow a connection with an issuance or proof request, which means as it is coded today then the user will be stuck on the screen waiting for the connection to complete (even though it has already completed). This can be configured differently using the config options in #365, however given that this is not the default leads to issues.
I think the current default behavior needs to after having connected and not having received an offer/request return a success screen or return the user to the home screen. 
The use case of immediately following a connection establishment with an offer or request will become easier with the usage of the `requests-attach` in the OOB invitations, but that's not immediately where we're at yet.
Thoughts? Happy to add a discussion topic to the next UG meeting agenda if helpful.

**Comment by alberto-instnt** [comment]
Im getting the same issue and stays on the loading screen and after 10 seconds I get a message about "taking longer than usual". It seems like it's waiting for changes on the `notificationRecord` in the` Connection.tsx`. There is no notification record coming in which means it's not transitioning to the next screen(credential). Ive tried navigating to the home screen if the record doesn't come in (same as in #365 ), but you wont necessarily see the notification right away. It may take a moment or even me having to reopen the app. This makes the apps UI/UX a bit iffy. Is there any other workaround solutions or techniques where I can go to home and see the credential notification? 
This seems to be only happening at the initial installation the app by the way.

**Comment by soubhratra-instnt** [comment]
This seems like a connectivity issue. There is no notificationRecord or invitation after scanning QR code. The app stalls on the "This is taking longer than usual" message. In order to see the actual invitation you need to restart the app and only then will you see the actual notification.

Here are the steps to reproduce this issue:
-Install and open App for the first time.
-Go through onboarding (terms, biometrics, etc.)
-Splash Screen (agent is initialized)
-Goes to main stack (Home).
-Press QR code and give app camera permissions.
-Scan QR code.
-Loading screen: "Just a moment while we make a secure connection...."
-Get "taking too long" message.

From here the app just stalls and we need to reset the app to get working properly. When the app gets killed and opened again, we see the VC notification and able to accept and store it.

We are using indicio mediator and bcgov testnet. Please let me know if any additional information is needed.

**Comment by JamesKEbert** [comment]
In your use case @soubhratra-instnt, are you doing a connection-less issuance, or connection followed by issuance? Also, if you navigate back to the home screen after the "taking too long" message appears are you able to see the connection (by default connections are listed in the settings in Bifold)? 
If your issuer is able, are you able to send or receive basic messages successfully (troubleshooting whether or not this is a connectivity issue or an issue with our handling around credential issuance in Bifold)?
If you have logs available that'd also be quite helpful!

**Comment by soubhratra-instnt** [comment]
Hi @JamesKEbert We are using connection-less issuance. When we go back to the home screen, and wait, we still don't see the VC but the connections shows up in the contact listing screen always irrespective of the failure to store the VC. We are not sure how to send / receive basic messages. We tried to use Aries toolbox but see a long list of device with bifold appearing many time as in the attached screenshot. We might need some guidance in this regard.

<img width="992" alt="image" src="https://user-images.githubusercontent.com/85633153/196524854-ca66b2d6-6660-4b26-bf6a-249a2bc488c7.png">

**Comment by JamesKEbert** [comment]
While a bit of a pain--the best way with the toolbox to ensure that you're using the right agent for issuance or basic messaging is to delete all the connections to "Aries Bifold" so that you only have one to choose from.

**Comment by shiva-instnt** [comment]
@JamesKEbert I am shiva working for instnt. When we initially send the message it gets delivered to the toolbox. But we don't see an acknowledgement on the "Aries Bifold". When we send the message back(the first message) from toolbox it does not appear on the bifold. But hereafter we see successful communication.

**Comment by soubhratra-instnt** [comment]
@JamesKEbert You can find the detail logs here https://github.com/hyperledger/aries-framework-javascript/issues/1048 where @TimoGlastra added some insights

**Comment by cvarjao** [comment]
closing stale issues
